docs/GettingStartedDocs/buildandsign.md

```diff
@@ -97,9 +97,37 @@ NumHeapPages=1024
 NumStackPages=1024
 NumTCS=1
 ```
+Additionally, a developer can specify additional Key Sharing and Separation (KSS) identity properties
```

Comment:
Updated

Reply:
I don't see any change.  I see a link in line 108 but it's not a link that defines KSS or says whether it's SGX only or TEE agnostic concept, or says whether it's the same or different from a "security domain".

Reply:
There do not seem to be any good docs defining KSS from what I can tell (the Intel manuals don't).   As near as I can tell, KSS is just the ability to share keys across multiple related enclaves, and in that sense the concept is not intel sgx specific, it's a subset of what the GlobalPlatform standard "security domain" provides (see https://globalplatform.org/specs-library/tee-management-framework-including-asn1-profile/ for the detailed spec), where as "security domain" contains the related enclaves, and "store data" within a "security domain" is equivalent to sealing with a family key in SGX KSS.   I could be wrong due to lack of sufficient docs, but that's my reading so far.  As such, we should make sure that OE can support the concept in general, while still mapping to Intel's functionality.   Question is whether we can have agnostic names now, or whether we need to use TEE (i.e., SGX) specific names now and possibly deprecate them later if agnostic names are defined, similar to what we did for report/quote changing to evidence in APIs.

Reply:
@dthaler No, KSS is not analogous to the functionality of Security Domains insofa; you should think of KSS as an opt-in extension of the existing SGX identity model by adding a couple of ID properties that will be included in the TEE-specific evidence. This includes:

- Two new sign-time authored identity values: FAMILYID and ISVEXTPRODID
- A new runtime identity value: CONFIGID, which is security versioned by CONFIGSVN.

If you squint at it enough, the comparison you're making kind of works, but really what this is about is providing a bigger identity namespace for SGX enclaves, and that doesn't intersect with the concept of permissions or key storage or association of keys to enclaves in a way that might relate to Security Domains and OTrP in GlobalPlatform.

All this does is allow a dev more options in providing differentiated identities for apps in a taxonomy of their choosing (which may not be hierarchical) Because this also translates into more ways to define an SGX sealing key based on which elements of the enclave identity are used in its derivation, it provides more flexibility for a dev to share/partition sealing keys, hence the name.

This example being SGX-specific I think is perfectly reasonable, and I don't believe this should be expected to generalize across TEEs, just as OP-TEE and SGX already do not share common identity metadata formats (i.e. OP-TEE basically only consisting of the TA's UUID)